src/main/java/org/opensearch/flowframework/transport/GetWorkflowTransportAction.java

```diff
@@ -0,0 +1,100 @@
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.client.Client;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.model.WorkflowState;
+import org.opensearch.flowframework.util.ParseUtils;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_STATE_INDEX;
+
+//TODO: Currently we only get the workflow status but we should change to be able to get the
+// full template as well
+/**
+ * Transport Action to get a specific workflow. Currently, we only support the action with _status
+ * in the API path but will add the ability to get the workflow and not just the status in the future
+ */
+public class GetWorkflowTransportAction extends HandledTransportAction<GetWorkflowRequest, GetWorkflowResponse> {
+
+    private final Logger logger = LogManager.getLogger(GetWorkflowTransportAction.class);
+
+    private final Client client;
+    private final NamedXContentRegistry xContentRegistry;
+
+    /**
+     * Intantiates a new CreateWorkflowTransportAction
+     * @param transportService the TransportService
+     * @param actionFilters action filters
+     * @param client The client used to make the request to OS
+     * @param xContentRegistry contentRegister to parse get response
+     */
+    @Inject
+    public GetWorkflowTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        NamedXContentRegistry xContentRegistry
+    ) {
+        super(GetWorkflowAction.NAME, transportService, actionFilters, GetWorkflowRequest::new);
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    protected void doExecute(Task task, GetWorkflowRequest request, ActionListener<GetWorkflowResponse> listener) {
+        String workflowId = request.getWorkflowId();
+        User user = ParseUtils.getUserContext(client);
+        GetRequest getRequest = new GetRequest(WORKFLOW_STATE_INDEX).id(workflowId);
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.get(getRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
+                logger.debug("Completed Get Workflow Status Request, id:{}", workflowId);
```

Comment:
If we want to have this debug statement, then it should be under the try block below

Reply:
removed actually